maskrcnn_benchmark/engine/trainer.py

```diff
@@ -107,6 +120,51 @@ def do_train(
             )
         if iteration % checkpoint_period == 0:
             checkpointer.save("model_{:07d}".format(iteration), **arguments)
+        if data_loaders_val is not None and test_period > 0 and iteration % test_period == 0:
+            synchronize()
+            for dataset_name, data_loader_val in zip(dataset_names, data_loaders_val):
+                _ = inference(  # The result can be used for additional loggin, e. g. to TensorBoard
+                    model,
+                    data_loader_val,
+                    dataset_name=dataset_name,
+                    iou_types=iou_types,
+                    box_only=False if cfg.MODEL.RETINANET_ON else cfg.MODEL.RPN_ONLY,
+                    device=cfg.MODEL.DEVICE,
+                    expected_results=cfg.TEST.EXPECTED_RESULTS,
+                    expected_results_sigma_tol=cfg.TEST.EXPECTED_RESULTS_SIGMA_TOL,
+                    output_folder=None,
+                )
+                synchronize()
+            model.train()
+            meters_val = MetricLogger(delimiter="  ")
+            with torch.no_grad():
+                for idx_val, (images_val, targets_val, _) in enumerate(data_loaders_val[0]):
+                    images_val = images_val.to(device)
+                    targets_val = [target.to(device) for target in targets_val]
+                    loss_dict = model(images_val, targets_val)
+                    losses = sum(loss for loss in loss_dict.values())
+                    loss_dict_reduced = reduce_loss_dict(loss_dict)
+                    losses_reduced = sum(loss for loss in loss_dict_reduced.values())
+                    meters_val.update(loss=losses_reduced, **loss_dict_reduced)
+            synchronize()
+            logger.info(
+                meters.delimiter.join(
```

Comment:
Should `meters` be `meters_val` ?

Reply:
It does not matter here because `meters` and `meters_val` have the same `delimiter`, but yes, ideally `meters_val` should be here. I'll fix this.